typing/ctype.ml

```diff
@@ -698,23 +698,24 @@ let duplicate_class_type ty =
    [expand_abbrev] (via [subst]) requires these expansions to be
    preserved. Does it worth duplicating this code ?
 *)
-let rec generalize ~level ty =
+let rec generalize ~outer_level ty =
   let ty_level = get_level ty in
-  if (ty_level > level) && (ty_level <> generic_level) then begin
+  if (ty_level > outer_level) && (ty_level <> generic_level) then begin
     set_level ty generic_level;
     (* recur into abbrev for the speed *)
     begin match get_desc ty with
       Tconstr (_, _, abbrev) ->
-        iter_abbrev (generalize ~level) !abbrev
+        iter_abbrev (generalize ~outer_level) !abbrev
     | _ -> ()
     end;
-    iter_type_expr (generalize ~level) ty
+    iter_type_expr (generalize ~outer_level) ty
   end
 
-(* Note: remove [level] argument at the end of this file *)
-let generalize ~level ty =
+(* Note: the public version of this function is called with
+   [ty] = [!current_level] *)
```

Comment:
`[~level]` = `[!current_level]`. Or maybe even shorter `[~level:!current_level]`...  Sorry.

Reply:
My bad, I should have been more careful!